dbt/task/deps.py

```diff
@@ -110,9 +111,19 @@ def __pull_deps_recursive(self, repos, processed_repos=None, i=0):
                     )
             except IOError as e:
                 if e.errno == errno.ENOENT:
-                    logger.info("'{}' is not a valid dbt project - "
-                                "dbt_project.yml not found".format(repo))
-                    exit(1)
+                    error_string = basestring(e)
+
+                    if 'dbt_project.yml' in error_string:
+                        error_string = ("'{}' is not a valid dbt project - "
+                                        "dbt_project.yml not found"
+                                        .format(repo))
+
+                    elif 'git' in error_string:
+                        error_string = ("Git CLI is a dependency of dbt, but "
+                                        "it is not installed!")
+
+                    raise dbt.exceptions.RuntimeException(error_string)
```

Comment:
@cmcarthur do you think there's any merit to always logging the raw error here? I'm a little afraid of masking the actual error, especially when the `if/elif` branches are traversed.